src/argilla/server/daos/backend/client_adapters/opensearch.py

```diff
@@ -554,23 +550,15 @@ def index_documents(
         return len(failed)
 
     @staticmethod
-    def _doc2bulk_action(
-        index: str,
-        doc: Dict[str, Any],
-    ) -> Dict[str, Any]:
-        def get_id(r):
-            return r.get("id")
+    def _doc2bulk_action(index: str, doc: Dict[str, Any]) -> Dict[str, Any]:
+        doc_id = doc.get("id")
 
-        data = {
-            "_op_type": "index",
-            "_index": index,
-            "_routing": None,  # TODO(@frascuchon): Use a sharding routing
-            **doc,
-        }
+        data = (
+            {"_index": index, "_op_type": "index", **doc}
+            if doc_id is None
```

Comment:
@frascuchon I believe we are currently creating doc ID Python client side, so this might have unexpected behaviour, i.e., the doc_id will never be None. https://github.com/argilla-io/argilla/issues/2340

Reply:
That means the upsert (update or insert) will be always applied. If the record already exists, will be updated with new info. If all the record info is sent, this approach still works for those cases